### PR TITLE
R4R: Construction of MsgEditToken and related test 

### PR DESCRIPTION
**Pull request by julia2804**
https://github.com/irisnet/irishub/issues/1652

### CODE REVIEW

File: client/asset/cli/flags.go [hunk 1]
```diff
@@ -68,4 +69,10 @@ func init() {
 	FsFeeQuery.String(FlagSubject, "", "the fee type to be queried")
 	FsFeeQuery.String(FlagMoniker, "", "the gateway name")
 	FsFeeQuery.String(FlagID, "", "the token id")
+
+	FsEditToken.String(FlagName, "", "the asset name, e.g. IRIS Network")
```
Comment (chengwenxi) [inline client/asset/cli/flags.go]: the token name, e.g. iris

Comment (chengwenxi) [inline client/asset/cli/flags.go]: Token name does not contain spaces.

File: client/asset/cli/flags.go [hunk 2]
```diff
@@ -68,4 +69,10 @@ func init() {
...
+	FsEditToken.String(FlagSymbolAtSource, "", "the source symbol of a gateway or external asset")
+	FsEditToken.String(FlagSymbolMinAlias, "", "the asset symbol minimum alias")
+	FsEditToken.Uint64(FlagMaxSupply, 0, "the max supply token of asset")
+	FsEditToken.Bool(FlagMintable, false, "whether the asset can be minted, default false")
```
Comment (chengwenxi) [inline client/asset/cli/flags.go]: All asset to token?

File: app/v1/asset/wire.go [hunk 3]
```diff
@@ -4,11 +4,12 @@ import (
 	"github.com/irisnet/irishub/codec"
 )
 
-// Register concrete types on codec
+// Register concrete types on codec codec
```
Comment (chengwenxi) [inline app/v1/asset/wire.go]: -- codec

File: app/v1/asset/msgs.go [hunk 4]
```diff
@@ -432,3 +432,88 @@ func ValidateMoniker(moniker string) sdk.Error {
...
+// ValidateBasic implements Msg
+func (msg MsgEditToken) ValidateBasic() sdk.Error {
+
+	//check owner
+	if msg.Owner.Empty() {
+		return ErrNilAssetOwner(DefaultCodespace, "the owner of the asset must be specified")
+	}
+
+	nameLen := len(msg.Name)
+	if nameLen > 0 && nameLen > MaximumAssetNameSize {
```
Comment (zhangyelong) [inline app/v1/asset/msgs.go]: if nameLen > 0 ?

File: app/v1/asset/msgs.go [hunk 5]
```diff
@@ -432,3 +432,88 @@ func ValidateMoniker(moniker string) sdk.Error {
...
+
+	//check symbol_at_source
+	symbolAtSourceLen := len(msg.SymbolAtSource)
+	if symbolAtSourceLen > 0 && (symbolAtSourceLen < MinimumAssetSymbolSize || symbolAtSourceLen > MaximumAssetSymbolSize || !IsAlphaNumeric(msg.SymbolAtSource)) {
+		return ErrInvalidAssetSymbolAtSource(DefaultCodespace, fmt.Sprintf("invalid token symbol_at_source %s, only accepts alphanumeric characters, length [%d, %d]", msg.SymbolAtSource, MinimumAssetSymbolSize, MaximumAssetSymbolSize))
+	}
+
+	//check symbol_min_alias
+	symbolMinAliasLen := len(msg.SymbolMinAlias)
+	if symbolMinAliasLen > 0 && (symbolMinAliasLen < MinimumAssetSymbolMinAliasSize || symbolMinAliasLen > MaximumAssetSymbolMinAliasSize || !IsAlphaNumeric(msg.SymbolMinAlias)) {
```
Comment (zhangyelong) [inline app/v1/asset/msgs.go]: symbolMinAlias can not begin with a `num`

File: docs/light-client/README.md [hunk 6]
```diff
@@ -146,7 +146,8 @@ Once IRISLCD is started, you can open `localhost:1317/swagger-ui/` in your explo
     2. `GET /asset/gateways`: Query all the gateways with an optional owner
     3. `POST /asset/gateways`: Create a gateway
     4. `PUT /asset/gateways/{moniker}`: Edit an existing gateway
-
+5. `PUT /asset/token/{token-id}`: Edit an existing token
```
Comment (zhangyelong) [inline docs/light-client/README.md]: /asset/token/{token-id} -> /asset/tokens/{token-id}

File: docs/zh/light-client/README.md [hunk 7]
```diff
@@ -148,7 +148,8 @@ IRISLCD启动以后，您可以在浏览器中访问`localhost:1317/swagger-ui/`
     2. `GET /asset/gateways`: 查询所有网关信息，提供一个可选的owner参数
     3. `POST /asset/gateways`: 创建一个网关
     4. `PUT /asset/gateways/{moniker}`: 编辑一个存在的网关
- 
+5. `PUT /asset/token/{token-id}`: 编辑一个已经存在的资产
```
Comment (zhangyelong) [inline docs/zh/light-client/README.md]: /asset/token/{token-id} -> /asset/tokens/{token-id}

File: lite/swagger-ui/swagger.yaml [hunk 8]
```diff
@@ -3387,6 +3387,84 @@ paths:
           description: Invalid id
         '500':
           description: Internal Server Error
+  '/asset/token/{token-id}':
```
Comment (zhangyelong) [inline lite/swagger-ui/swagger.yaml]: /asset/token/{token-id} -> /asset/tokens/{token-id}

**Pull request merged**